msctl

```diff
@@ -1028,29 +1038,36 @@ start() {
   rotateLog "$1"
   # Make a mirror image of the world directory if requested.
   if [ $ENABLE_MIRROR -eq 1 ]; then
-    mkdir -p "$MIRROR_PATH/$1"
-    if [ $? -ne 0 ]; then
-      printf "Error copying world data, path %s not found.\n" $MIRROR_PATH/$1
-      exit 1
-    fi
-    # Check for a clean dismount from the previous server run.  If we have a
-    # <world>-original directory within <world> we didn't stop cleanly.
-    if [ -d "$WORLDS_LOCATION/$1/$1-original" ]; then
-      # Remove the symlink to the world-file mirror image.
-      # if recently restored from backup, this symlink may or may not exist
-      if [ -L "$WORLDS_LOCATION/$1/$1" ]; then
-        rm -r "$WORLDS_LOCATION/$1/$1"
+    # Make sure the mirror master folder exists.
+    mkdir -p "$MIRROR_PATH"
+    # If the symlink exists but the target doesn't, the
+    # mirror was removed somehow. This is bad.
+    if [ -L "$WORLD_DIR/$1" ] && [ ! -d "$MIRROR_PATH/$1" ]; then
+      # Remove the symlink to the mirror image.
+      rm -f "$WORLD_DIR/$1"
+      # Restore the original world files.
+      mv "$WORLD_DIR/$1-original" "$WORLD_DIR/$1"
+      # Send notification.
+      printf "Warning, the mirror for %s was removed, restored latest world.\n" $1
+    fi;
+    # If the symlink still exists, the server was stopped outside of mscs.
+    # SO DONT RESET THE MIRROR, just keep using it.
+    if [ ! -L "$WORLD_DIR/$1" ]; then
+      # Remove the mirror directory just in case.
+      rm -Rf "$MIRROR_PATH/$1"
+      # Copy the world files over to the mirror.
+      cp -a "$WORLD_DIR/$1" "$MIRROR_PATH/$1"
+      if [ $? -ne 0 ]; then
+        printf "Error copying world data, could not copy to %s.\n" $MIRROR_PATH/$1
```

Comment:
Should some additional help text be added so users know what they can do to fix this?